pkg/api/testapi/testapi.go

```diff
@@ -115,7 +115,11 @@ func ResourcePath(resource, namespace, name string) string {
 // /api/v1beta1/pods/pod0?namespace=foo for v1beta1 and
 // /api/v1beta3/namespaces/foo/pods/pod0 for v1beta3.
 func ResourcePathWithQueryParams(resource, namespace, name string) string {
-	path := ResourcePath(resource, namespace, name)
+	return ResourcePathWithPrefixAndQueryParams("", resource, namespace, name)
+}
+
+func ResourcePathWithPrefixAndQueryParams(prefix, resource, namespace, name string) string {
```

Comment:
The name of this function is misleading. It would be great if it handled appending query params.